extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -275,6 +296,14 @@ export class AmpAd3PImpl extends AMP.BaseElement {
       user.assert(!this.isInFixedContainer_,
           '<amp-ad> is not allowed to be placed in elements with ' +
           'position:fixed: %s', this.element);
+      /** detect ad containers, add the list to element as a new attribute */
+      if (!this.element.getAttribute('data-amp-container-element')) {
+        const containerList_ = getAdContainerList(this.element);
```

Comment:
This isn't a list, though. And, no need for the `_` suffix, this isn't attached to the object.